### PR TITLE
Updates splash screen package

### PR DESCRIPTION
**Pull request by burkeshartsis**
### What does this PR do, or why is it needed?
This PR switches the package we use to manage splash screens and migrates us away from `react-native-splash-screen` to `react-native-bootsplash`. The old package was abandoned, didn't support rn60 so wasn't fully working on ios, and [an announcement from apple](https://developer.apple.com/news/?id=01132020b) required everyone to change how "launch/splash" screens were implemented. The new package checks all the boxes. 
 
### What design trade-offs/decisions were made?
Some times in the simulator I notice a slight flicker during launch on ios. It's not all the time but does show up in the gif above. The maintainer is tracking this or a similar issue [here](https://github.com/zoontek/react-native-bootsplash/issues/59).

Additionally, this PR has a lot of file changes but its mostly file name changes for image assets and their respective platform references. The actual footprint of this PR is small and if you aren't super familiar with the native implementations for this feature the file changes aren't going to make a ton of sense anyways. You are welcome to look over the packages installation instructions to check the relevant file changes.

### How do I test this PR?
1. Launch app.
2. 👍 

Some things worth considering:
- Android
  - I wasn't completely sure where to put the `launchMode`. Based on the information [here](https://github.com/zoontek/react-native-bootsplash#handle-deep-linking-on-android) I added it to both activities. I didn't, however, add the second `intent-filter` that react-native-bootsplash says you need for deeplinking to work. I think we should test if it works without and then add it if we do. @vinnyjth maybe you will know more about the `android:scheme` it says we'll need for deeplinks.

### What automated tests did you write?
This is all native code for the most part. Nothing to write tests for.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by conrad-vanl** [comment]
@burkeshartsis do you know if it's possible to run `yarn generate-bootsplash` multiple times? Like if you wanted to change the splash screen used? Should we add that command to our `package.json` in some form so it's easier for developers to discover?

File: apolloschurchapp/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -14,21 +14,30 @@
...
         android:name="com.google.android.geo.API_KEY"
         android:value="@string/GOOGLE_MAPS_API_KEY"/>
+
       <activity
         android:name=".MainActivity"
         android:label="@string/app_name"
         android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
         android:windowSoftInputMode="adjustPan"
+        android:exported="true"
+        android:launchMode="singleTask">
```
Comment (conrad-vanl) [inline apolloschurchapp/android/app/src/main/AndroidManifest.xml]: Does this have any side affects we know of?

File: apolloschurchapp/index.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
-import './loadConfig';
-import { AppRegistry, YellowBox } from 'react-native';
-import SplashScreen from 'react-native-splash-screen';
-import ApollosConfig from '@apollosproject/config';
-import Storybook from './storybook';
+import "./loadConfig";
+import { AppRegistry, YellowBox } from "react-native";
+// import SplashScreen from 'react-native-splash-screen';
```
Comment (conrad-vanl) [inline apolloschurchapp/index.js]:
```suggestion
```

File: apolloschurchapp/index.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
-import './loadConfig';
-import { AppRegistry, YellowBox } from 'react-native';
-import SplashScreen from 'react-native-splash-screen';
-import ApollosConfig from '@apollosproject/config';
-import Storybook from './storybook';
+import "./loadConfig";
```
Comment (conrad-vanl) [inline apolloschurchapp/index.js]: Did eslint get messed up? Why are all of the single quotes `'` changing to double quotes `"`?

File: apolloschurchapp/index.js [hunk 4]
```diff
@@ -17,11 +17,11 @@ if (useStorybook) {
 // This tries to hide the splash screen so you can see the error.
 // Shouldnt't do anything in Prod.
 // Depending on the error, splash screen might not go away ;)
```
Comment (conrad-vanl) [inline apolloschurchapp/index.js]:
```suggestion
```

File: apolloschurchapp/index.js [hunk 5]
```diff
@@ -17,11 +17,11 @@ if (useStorybook) {
...
-global.ErrorUtils.setGlobalHandler(() => SplashScreen.hide());
+// global.ErrorUtils.setGlobalHandler(() => SplashScreen.hide());
```
Comment (conrad-vanl) [inline apolloschurchapp/index.js]:
```suggestion
```

File: apolloschurchapp/src/index.js [hunk 6]
```diff
@@ -1,39 +1,40 @@
-import hoistNonReactStatic from 'hoist-non-react-statics';
-import React from 'react';
-import { StatusBar } from 'react-native';
-import { createStackNavigator, createAppContainer } from 'react-navigation';
-import SplashScreen from 'react-native-splash-screen';
+import hoistNonReactStatic from "hoist-non-react-statics";
```
Comment (conrad-vanl) [inline apolloschurchapp/src/index.js]: Same with the `'` to `"` change?

File: apolloschurchapp/src/index.js [hunk 7]
```diff
@@ -1,39 +1,40 @@
...
-  barStyle: 'dark-content',
-  backgroundColor: theme.colors.paper,
+  barStyle: "dark-content",
+  backgroundColor: theme.colors.paper
 }))(StatusBar);
 
-const ProtectedRouteWithSplashScreen = (props) => {
-  const handleOnRouteChange = () => SplashScreen.hide();
+const ProtectedRouteWithSplashScreen = props => {
+  const handleOnRouteChange = () => RNBootSplash.hide({ duration: 250 });
```
Comment (conrad-vanl) [inline apolloschurchapp/src/index.js]: is `250` a "magic" number?

File: apolloschurchapp/storybook/index.js [hunk 8]
```diff
@@ -1,21 +1,22 @@
...
 addApollosProviderDecorator(Providers);
 
 configure(() => {
-  require('./Welcome');
+  require("./Welcome");
   loadStories();
   loadApollosStories();
 
-  SplashScreen.hide();
+  RNBootSplash.hide({ duration: 250 });
```
Comment (conrad-vanl) [inline apolloschurchapp/storybook/index.js]: Same comment on the `250`?

**Comment by burkeshartsis** [comment]
@conrad-vanl you can run it as many times as you want but it wipes out the old assets each time. It's also an interactive CLI so you can't script it to automatically run within user interaction. You can see the corresponding feature request [here](https://github.com/zoontek/react-native-bootsplash/issues/51).

File: apolloschurchapp/src/index.js (hunk 7 above)
Comment (burkeshartsis) [reply apolloschurchapp/src/index.js]: It's milliseconds for the fade animation. So semi-magical?

**Comment by conrad-vanl** [comment]
> you can run it as many times as you want but it wipes out the old assets each time.

Is this a problem?

> so you can't script it to automatically run within user interaction

Yeah, I think that's fine. I'm just wanting to make sure that developers starting their own app have visibility to this command. At the least, we should add it to our (very oudated) readme.md 🤷‍♂️ But thought adding it to our `package.json` as like a `generate-splashscreen` command or something could help discovery as well.

File: apolloschurchapp/src/index.js (hunk 7 above)
Comment (conrad-vanl) [reply apolloschurchapp/src/index.js]: is the value required?

**Comment by conrad-vanl** [comment]
@burkeshartsis I just merged one of the PRs that change the eslint setup some - try merging this with master, yarning, and re linting and see if it fixes these lint changes

File: apolloschurchapp/src/index.js (hunk 7 above)
Comment (burkeshartsis) [reply apolloschurchapp/src/index.js]: nope! Just looks prettier with it.

File: apolloschurchapp/android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (burkeshartsis) [reply apolloschurchapp/android/app/src/main/AndroidManifest.xml]: So `launchMode` I called out in the PR description. We had it before and will continue to need it. I updated our usage to match what this package says we'll need for deeplinking to work. 

As for `exported` this was an add based on direction from `react-native-bootsplash`. Having just read [the documentation](https://developer.android.com/guide/topics/manifest/activity-element#exported) for it now it sounds like we probably should have already had this value set to `true` based on the suggestions in the Android docs. I think the package was just following best practices better than we were previously. To answer you question @conrad-vanl, I don't think there are any **unwanted** consequences given that we were already using `intent-filters` for deeplinks and push notifications. A positive is that now, other apps can "link into"/launch our app.

File: apolloschurchapp/index.js (hunk 3 above)
Comment (burkeshartsis) [reply apolloschurchapp/index.js]: My prettier setup is set to follow our eslint setup. Since eslint is apparently not setup prettier is doing whatever it wants 🤦‍♂ I've not set prettier to only make changes if eslint is setup but we'll still need to fix eslint in a different PR.

**Comment by burkeshartsis** [comment]
I think this is ready for review again.
